app/controllers/orders_controller.rb

```diff
@@ -1,13 +1,24 @@
 class OrdersController < ApplicationController
+  before_action :authenticate_user!
+
   def add_to_order
-    if Order.last
-      order = Order.last
-    else
-      order = Order.create
+	  if session[:order_id] == nil || Order.find(session[:order_id]).finalized == true
```

Comment:
I have problem following this logic but nonetheless, this line can be simplified
```
if !session[:order_id] || Order.find(session[:order_id]).finalized
```
But again, I think you can omit the last check and consequently use 
```
unless session[:order_id]
```